manifest/schema2/manifest.go

```diff
@@ -48,6 +52,14 @@ func init() {
 	}
 }
 
+// LayerDescriptor is an extension to distribution.Descriptor that
+// has additional fields specific to layers.
+type LayerDescriptor struct {
```

Comment:
I had a feeling that we wouldn't want the `URL` concept to pervade all the other places where we use descriptors to reference internal resources, but I'm okay with including the field in `Descriptor` if @stevvooe is.